lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/NoCredentialsAuth.java

```
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.gcs;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.NoCredentials;
+import com.google.cloud.storage.StorageOptions;
+import io.trino.spi.security.ConnectorIdentity;
+
+import java.io.IOException;
+
+public class NoCredentialsAuth
+        implements GcsAuth
+{
+    @Override
+    public void setAuth(StorageOptions.Builder builder, ConnectorIdentity identity)
+    {
+        try {
+            builder.setCredentials(GoogleCredentials.getApplicationDefault());
+        }
+        catch (IOException e) {
+            builder.setCredentials(NoCredentials.getInstance());
+        }
```

Can we just leave `builder.setCredentials(NoCredentials.getInstance());`?
The `NoCredentialsAuth` class single responsibility is to provide the UNAUTHENTICATED access only, no?

IMO, we do not need `builder.setCredentials(GoogleCredentials.getApplicationDefault())` here because it breaks the purpose and it is somehow misleading. I'd leave it in [GcsServiceAccountAuth](https://github.com/trinodb/trino/pull/27106/files#diff-0a10de45766835578f6e53acf7c91c94cf85858a6ef42e2f36badbc49dd50616) or extract it to a dedicated eg `GcsAdcAuth` class (potential breaking change).

See my original (but I think outdated) proposal/idea: https://github.com/trinodb/trino/issues/26854#issuecomment-3374313658

---

Make sense, I complicated the problem, switched